### DISCUSSION TITLE
Blazor multiple tabs issue with IIS (fine with IIS Express)

### DISCUSSION THREAD

**Discussion by michalkubis** [dotnet/aspnetcore#24501] [General] [answered]
IIS Express works as expected, but when I use IIS I have a problem with multiple tabs. See below:
![blazor1](https://user-images.githubusercontent.com/47777142/89097966-d999e580-d3e3-11ea-8bb9-a67eddf7373a.png)
![blazor2](https://user-images.githubusercontent.com/47777142/89097968-ddc60300-d3e3-11ea-8bc0-9b1463b4922c.png)
![blazor3](https://user-images.githubusercontent.com/47777142/89097971-e1598a00-d3e3-11ea-9caa-840e6f1940c5.png)

2nd, or 3rd tab usually fails to open menu icons.
4rd tab always shows just blank page with Loading... tab (with status Waiting for localhost).
I open other tabs right clicking on the app's title "Open link in new tab". 
Duplicate tab does the same.

When I close single tab, the other (problematic one) will load just fine immediately.

Could someone help me to understand what's going on and why it does not work on IIS, but only in IIS Express? Thank you in advance.

**Comment by davidfowl** [comment, marked as answer]
Assuming this is blazor server? Windows client apps have a maximum websocket limit on IIS that you might be hitting.

**Reply by michalkubis** [reply]
Yes, it is blazor server. It's working fine with IIS Express.

**Reply by davidfowl** [reply]
Yep, that's by design unfortunately. IIS has a concurrent request limit on the client OS.

**Reply by michalkubis** [reply]
When I deployed it to IIS on WS2016 everything's fine. 
I don't understand, it must be my IIS settings, or I'm missing something.

EDIT:
I've just read this https://stackoverflow.com/questions/62262471/blazor-server-on-iis10-only-4-concurrent-connections

Is there a way to expand this limit? That's quite odd behaviour and it's hard to find a word about it.

**Reply by davidfowl** [reply]
Not on IIS no. You can use another webserver, like kestrel.

**Reply by michalkubis** [reply]
Thank you.
